### DISCUSSION TITLE
Is there an option to limit CPU core count / usages?

### DISCUSSION THREAD

**Discussion by kwonoj** [web-infra-dev/rspack#8276] [Q&A]
Trying to make migration poc for the large code base and due to some legacy codebase we are experiencing initial build takes some time. It is fine, but at some stage of build rspack seems excessively occupies CPU (or ram) usage makes other system activites feel hang - literally whole UI stops and hourglass icon appears around 20~30sec.

I'm curious if there's a way to set the limit for the CPU core count or similar, so it may take longer but won't block system use while dev waiting for it.

**Comment by hardfist** [comment]
try set `TOKIO_WORKER_THREADS` and `RAYON_NUM_THREADS`, in the future we may support `RSPACK_WORKER_THREADS` to control both

**Reply by kwonoj** [reply]
I gave it a try with `TOKIO_WORKER_THREADS=4 RAYON_NUM_THREADS=4 rspack serve`, but still observing same thing. Is there a way to diagnose this?

**Reply by kwonoj** [reply]
Ok, trying a bit more it looks like it's not a fault of rspack. If I use `sass-loader` with implementation `sass-embedded`, it seems spawning huge numbers of process at once (we have > 2000 scss modules) and it chokes up the CPU. Could workaround by not using sass-embedded, but would be good if rspack could throttle the number of the process spawned by loader or plugins.
